esrally/mechanic/mechanic.py

```diff
@@ -39,6 +41,122 @@ def download(cfg):
     console.println(json.dumps(binaries, indent=2), force=True)
 
 
+def provision(cfg):
+    challenge_root_path = paths.race_root(cfg)
+    car, plugins = load_team(cfg, False)
+
+    s = supplier.create(cfg, sources=False, distribution=True, build=False,
+                        challenge_root_path=challenge_root_path, car=car, plugins=plugins)
+
+    all_node_ips = cfg.opts("provisioning", "node.ips", mandatory=False, default_value=[])
```

Comment:
There is one issue here; since `node.ips` is not mandatory, if not specified `all_node_ips` is an empty list.
The `elasticsearch.yml` templates in [rally-teams](https://github.com/elastic/rally-teams/blob/b5c15819f41dc7471610a04d74487636183b73eb/cars/v1/vanilla/templates/config/elasticsearch.yml#L68-L81) rely on the presence of this and if defined but empty this gives me an elasticsearch.yml discovery looking like:

```
discovery.seed_hosts: ["]
# Prevent split brain by specifying the initial master nodes.
cluster.initial_master_nodes: [""]
```

Currently, the single node use case provisions with:
```
discovery.seed_hosts: ["127.0.0.1"]
# Prevent split brain by specifying the initial master nodes.
cluster.initial_master_nodes: ["127.0.0.1"]
```